Additions to Chatbot Pattern (#446)
carbon-design-system/carbon-website: src/pages/experimental/chatbot/overview.mdx

---
title: Chatbots
description: Use a chatbot when a user would benefit from requesting information on-demand with natural language. They should receive a useful and trustworthy response related to their question or task.
tabs: ['Overview', 'Usage', 'Flows', 'Content']
---

import { Tag, Link } from 'carbon-components-react';

### Use a chatbot when a user would benefit from requesting information on-demand with natural language. They should receive a useful and trustworthy response related to their question or task.

> [Download the Sketch Kit](#chatbot-add-on-library)

Reviewer: I feel like the link sitting alone here is a little weird. I think the tile slightly further down the page should be fine.

While conversational interfaces may contain many different kinds of components, Carbon currently offers these components out of the box to help get you up and running quickly. See the Usage tab for information on how to use these components together.

> | Component             | Design Status                            | Development Status                               |

Reviewer: ```suggestion
| Component             | Design status                            | Development status                               |
```

| System message        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution needed</Tag> |
| User message          | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution needed</Tag> |
| Structured response   | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution needed</Tag> |
| Chatbot cards         | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution needed</Tag> |
| Chatbot text input    | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution needed</Tag> |
| Launch button         | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution needed</Tag> |
| Welcome screen        | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution needed</Tag> |
| Expandable card       | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution needed</Tag> |
| Human message         | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution needed</Tag> |
| List                  | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |

> | Launch Button chips   | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |

Reviewer: ```suggestion
| Launch button chips   | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |
```

| Multiple threads menu | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |

## Chatbot add-on library

> To **add this library to Sketch with Sketch Cloud**, access the link below and click "Add Library to Sketch" button. You will receive updates to chatbot components automatically.

Reviewer: ```suggestion
To **add this library to Sketch with Sketch Cloud**, access the link below and click "Add Library to Sketch." You will receive updates to chatbot symbols automatically.
```

> For **common chatbot sub-patterns, features or experimental components**, click the "Download Document" button, open the document, and browse the relevant pages for guidance related to your use case.

Reviewer: ```suggestion
For **common chatbot sub-patterns, features or experimental components**, click "Download Document," open the document, and browse the relevant pages for guidance related to your use case.
```

While conversational interfaces may contain many different kinds of components, Carbon currently offers these components out of the box to help get you up and running quickly. See the Usage tab for information on how to use these components together.

| Component             | Design Status                            | Development Status                               |
| --------------------- | ---------------------------------------- | ------------------------------------------------ |

> | Chatbot Window        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Chatbot window        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Chatbot Header        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Chatbot header        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | System Message        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| System message        | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

| User Message          | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

> | Structured Response   | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Structured response   | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Chatbot Cards         | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Chatbot cards         | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Chatbot Text Input    | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Chatbot text input    | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Launch Button         | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Launch button         | <Tag type="green">Design available</Tag> | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Welcome Screen        | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Welcome screen        | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Expandable Card       | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Expandable card       | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution Needed</Tag> |
```

> | Human Message         | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Human message         | <Tag type="teal"> Experimental </Tag>    | <Tag type="cool-gray"> Contribution needed</Tag> |
```

> | List                  | <Tag type="purple"> Coming Soon </Tag>   | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| List                  | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |
```

> | Launch Button Chips   | <Tag type="purple"> Coming Soon </Tag>   | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Launch button chips   | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |
```

> | Multiple Threads Menu | <Tag type="purple"> Coming Soon </Tag>   | <Tag type="cool-gray"> Contribution Needed</Tag> |

Reviewer: ```suggestion
| Multiple threads menu | <Tag type="purple"> Coming soon </Tag>   | <Tag type="cool-gray"> Contribution needed</Tag> |
```
